containers/src/Data/Set/Internal.hs

```diff
@@ -906,19 +1140,33 @@ foldr :: (a -> b -> b) -> b -> Set a -> b
 foldr f z = go z
   where
     go z' Tip           = z'
-    go z' (Bin _ x l r) = go (f x (go z' r)) l
+    go z' (NE (Bin' _ x l r)) = go (f x (go z' r)) l
 {-# INLINE foldr #-}
 
+foldr1 :: (b -> b -> b) -> NonEmptySet b -> b
```

Comment:
Follow the discussion in https://mail.haskell.org/pipermail/libraries/2019-October/030040.html., maybe this method should be called `semifoldr` in the future. (We'll hopefully know the final name in two weeks).

Also having

```haskell
semifoldrMap :: (a -> b) -> (a -> b -> b) -> NonEmptySet a -> b
```

would be beneficial, if that's possible to implement easily.

Reply:
@phadej, I like your `semifoldrMap` quite a bit, but its argument order is confusingly different from `foldr`'s.

Reply:
Do see what the thread said, @treeowl. In particular `Semigroup => Group` but `Foldable => Semifoldable`. While I agree the `1` suffix is vague, I think using the Semi prefix here is more confusing.

Reply:
@Ericson2314, I was expressing an opinion on the type, not the name. Did the discussion reach a conclusion?

Reply:
Ah OK. I'm not sure, but we did include a function like that in this PR for the moment for both internal use and being able to write that instance in semigroupioids.